diff --git a/svf-llvm/lib/CHGBuilder.cpp b/svf-llvm/lib/CHGBuilder.cpp
index 1cea133f..ca2a9ca9 100644
--- a/svf-llvm/lib/CHGBuilder.cpp
+++ b/svf-llvm/lib/CHGBuilder.cpp
@@ -96,6 +96,11 @@ void CHGBuilder::buildCHGNodes(const GlobalValue *globalvalue)
         if (!chg->getNode(className))
             createNode(className);
 
+        if (vtblStruct->getNumOperands() == 2 &&
+            SVFUtil::isa<ConstantStruct>(vtblStruct->getOperand(0)) &&
+            vtblStruct->getOperand(1)->getType()->isArrayTy())
+            vtblStruct = SVFUtil::cast<ConstantStruct>(vtblStruct->getOperand(0));
+
         for (unsigned int ei = 0; ei < vtblStruct->getNumOperands(); ++ei)
         {
             const ConstantArray *vtbl = SVFUtil::dyn_cast<ConstantArray>(vtblStruct->getOperand(ei));
@@ -377,6 +382,11 @@ void CHGBuilder::analyzeVTables(const Module &M)
 
             node->setVTable(LLVMModuleSet::getLLVMModuleSet()->getSVFGlobalValue(globalvalue));
 
+            if (vtblStruct->getNumOperands() == 2 &&
+                SVFUtil::isa<ConstantStruct>(vtblStruct->getOperand(0)) &&
+                vtblStruct->getOperand(1)->getType()->isArrayTy())
+                vtblStruct = SVFUtil::cast<ConstantStruct>(vtblStruct->getOperand(0));
+
             for (unsigned int ei = 0; ei < vtblStruct->getNumOperands(); ++ei)
             {
                 const ConstantArray *vtbl =
